### ISSUE TITLE
Plot.AxisPanCenter() fails on non-primary axes

### ISSUE THREAD

**Issue by dusko23** [ScottPlot/ScottPlot#2483]
**Scenario:**
3 plots using Primary YAxis (representing various Altitudes)
3 plots using Secondary Yaxis - YAxis2 (representing various Speeds)
All 6 plots use a **common Primary Xaxis** which represents Distance.

**The plan:**
When the relevant cell in DataGridView is clicked ScottPlot should center there and draw a circle
![image](https://user-images.githubusercontent.com/26789352/224515775-8b241c9c-4019-43ed-b59c-71de7cf53571.png)

**The code:**
![image](https://user-images.githubusercontent.com/26789352/224515464-f47e4aa0-d716-45ec-be56-7e19546697ef.png)

Parameters **X**, **Y**, and **YaxisINDEX** are populated correctly.
For the first 3 plots YaxisINDEX = 0
For the next 3 plots YaxisINDEX = 1

**Bug Description:**
Running **The Code** YaxixINDEX = 1 has the effect of being = 0
**The plot is centered and the circle is drawn at the wrong Y position.**
**The Y pixel position to Center and Draw Marker  is calculated from YAxis instead of YAxis2.** 
_Please note that AddMarker does not expose the Y axis. But it should._

**Comment by swharden** [comment]
> Please note that AddMarker does not expose the Y axis. But it should.

I think you can do something like

```cs
var mark = pathPLOT.Plot.AddMarker(/* */)
mark.YAxisIndex = 1;
```

and it will tell your marker which axis the Y position represents.

Is this sufficient to solve your problem? If not, open this issue again and we can take a closer look 👍

**Comment by dusko23** [comment]
I am 100% sure there is a bug centering to Y secondary axis,
The marker is always drawn.
However, the center sometimes works, sometimes is offset a bit, and sometimes the center graph is shown completely wrong. When I pan to the point that should be centered in the first place marker is correctly shown.

I have carefully checked my code. It is the same as for the primary Y axis which works without any problems.
Changing Y axis index to 1 causes all of the grief :-)

**Comment by dusko23** [comment]
The above happens in zoom-in scenarios.
I am aware if Y limits are set to min-max, it can happen that center can not be shown.
However, if the center can not be shown plot should move to that point (below or above the correct center point). 
The same applies to the X-axis if the point to be centered is too close to either X-axis limit.

**Comment by swharden** [comment]
> I am 100% sure there is a bug centering to Y secondary axis,
> The marker is always drawn.

Thanks for clarifying this @dusko23! It's hard to have this conversation without a code example that demonstrates this issue. I'll try to prepare something to help move this conversation forward

**Comment by dusko23** [comment]
Here is a code example of causing grief. In VB.
![image](https://user-images.githubusercontent.com/26789352/229319704-2d1a4a66-edda-4c10-981e-b2b234361408.png)

If YaxisINDEX= 0 everything works just fine.
if  YaxisINDEX= 1 (Yaxis2 it is) I have problems when Plot is zoomed in.

**Comment by swharden** [comment]
Hi @dusko23, thank you for the screenshot! Unfortunately I can't execute that code to demonstrate this problem or evaluate if a code modification I make is able to fix it 😅 

Here is a console app that demonstrates the issue you describe. We can use this to try to locate the source of this problem 👍 

```cs
ScottPlot.Plot plt = new(400, 300);

// All 6 plots use a common Primary Xaxis
var mark1 = plt.AddMarker(1, 1);
var mark2 = plt.AddMarker(2, 2);
var mark3 = plt.AddMarker(3, 3);
var mark4 = plt.AddMarker(4, 4);
var mark5 = plt.AddMarker(5, 5);
var mark6 = plt.AddMarker(6, 6);

// 3 plots using Primary YAxis
mark1.YAxisIndex = 0;
mark2.YAxisIndex = 0;
mark3.YAxisIndex = 0;

// 3 plots using Secondary Yaxis
mark4.YAxisIndex = 1;
mark5.YAxisIndex = 1;
mark6.YAxisIndex = 1;
plt.YAxis2.Ticks(true);

// show the original axis limits
plt.SetAxisLimits(0, 10, 0, 10, yAxisIndex: 0);
plt.SetAxisLimits(0, 10, 0, 10, yAxisIndex: 1);
plt.SaveFig("dusko.png");

// put a mark at a new position and center on it
plt.AddMarker(2.5, 2.5, ScottPlot.MarkerShape.asterisk);
plt.AxisPanCenter(2.5, 2.5, yAxisIndex: 0);
plt.SaveFig("dusko2.png");

// trying to center on a non-primary axis demonstrates the issue
plt.AxisPanCenter(2.5, 2.5, yAxisIndex: 1);
plt.SaveFig("dusko3.png");
```

dusko.png | dusko2.png | dusko3.png
---|---|---
![image](https://user-images.githubusercontent.com/4165489/229320150-ba6c24aa-fa2b-403a-a4fe-0110878f1218.png)|![image](https://user-images.githubusercontent.com/4165489/229320127-27ee4443-557a-4a53-87b3-218162ae223d.png)|![image](https://user-images.githubusercontent.com/4165489/229320135-d6789bff-5118-46c8-8766-2e9fe3a93190.png)

**Comment by dusko23** [comment]
To reproduce the error I am referring to you need to zoom in. Try to zoom and pan to the area where nothing is shown. Then try to center to a specific point using Yaxix2. As well, at startup set outer limits based on the union of all data.
In my case, I am not using the function SetAxisLimits. Just outer limits. 
If it makes sense to you I can prepare a full-blown solution in VB to demonstrate the problem.

**Comment by swharden** [comment]
Thanks for the additional information @dusko23! I think I resolved the issue in #2544 and the example code above now produces this output:

![image](https://user-images.githubusercontent.com/4165489/229322919-b3d78a64-3d82-48a7-a81a-14aa0a750122.png)

I'll merge this in and publish a new package on NuGet today or tomorrow. If it doesn't fix your problem, let me know and we can re-open this issue and take a closer look!

Thanks again for reporting this.
